packages/mui-material/src/md2/index.ts

```diff
@@ -0,0 +1,357 @@
+          },
+          style: {
+            padding: '7px 21px',
+            fontSize: theme.typography.pxToRem(15),
+          },
+        },
+        {
+          props: {
+            size: 'small',
+            variant: 'contained',
+          },
+          style: {
+            padding: '4px 10px',
+            fontSize: theme.typography.pxToRem(13),
+          },
+        },
+        {
+          props: {
+            size: 'large',
+            variant: 'contained',
+          },
+          style: {
+            padding: '8px 22px',
+            fontSize: theme.typography.pxToRem(15),
+          },
+        },
+        {
+          props: {
+            disableElevation: true,
+          },
+          style: {
+            boxShadow: 'none',
+            '&:hover': {
+              boxShadow: 'none',
+            },
+            [`&.${buttonClasses.focusVisible}`]: {
+              boxShadow: 'none',
+            },
+            '&:active': {
+              boxShadow: 'none',
+            },
+            [`&.${buttonClasses.disabled}`]: {
+              boxShadow: 'none',
+            },
+          },
+        },
+        {
+          props: { fullWidth: true },
+          style: { width: '100%' },
+        },
+      ],
+    };
+  }),
+);
+
+const commonIconStyles = [
+  {
+    props: { size: 'small' },
+    style: {
+      '& > *:nth-of-type(1)': {
+        fontSize: 18,
+      },
+    },
+  },
+  {
+    props: { size: 'medium' },
+    style: {
+      '& > *:nth-of-type(1)': {
+        fontSize: 20,
+      },
+    },
+  },
+  {
+    props: { size: 'large' },
+    style: {
+      '& > *:nth-of-type(1)': {
+        fontSize: 22,
+      },
+    },
+  },
+];
+const ButtonMd2StartIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'StartIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.startIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})({
+  display: 'inherit',
+  marginRight: 8,
+  marginLeft: -4,
+  variants: [
+    {
+      props: { size: 'small' },
+      style: {
+        marginLeft: -2,
+      },
+    },
+    ...commonIconStyles,
+  ],
+});
+
+const ButtonMd2EndIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'EndIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.endIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})({
+  display: 'inherit',
+  marginRight: -4,
+  marginLeft: 8,
+  variants: [
+    {
+      props: { size: 'small' },
+      style: {
+        marginRight: -2,
+      },
+    },
+    ...commonIconStyles,
+  ],
+});
+
+const slots = {
+  MuiButton: {
+    variant: 'contained' as const,
+    color: 'primary' as const,
+    size: 'medium' as const,
+    slots: {
+      root: ButtonMd2Root,
+      startIcon: ButtonMd2StartIcon,
+      endIcon: ButtonMd2EndIcon,
+    },
+  },
+  // MuiTextField: {
+  //  slots: {
+  //    root: TextFieldRoot,
+  //  },
+};
```

Comment:
Yes, that I would use! https://www.notion.so/mui-org/material-ui-Post-v6-future-5960b255de7f4f418158fedc1dca4cee?pvs=4#6bccd223cbc14985b1c8e7dcbf66edf0

```suggestion
// @mui/md3
import styled from '@mui/system';
import BaseButton from '@mui/base/BaseButton';

const buttonSlots = {
  root: ButtonMd2Root,
  startIcon: ButtonMd2StartIcon,
  endIcon: ButtonMd2EndIcon,
};

export const Button = React.forwardRef((props, ref) => {
  <BaseButton slots={buttonSlots} {...props} />
})
```